llama-index-integrations/vector_stores/llama-index-vector-stores-lancedb/llama_index/vector_stores/lancedb/base.py

```diff
@@ -88,34 +119,77 @@ class LanceDBVectorStore(BasePydanticVectorStore):
         ```python
         from llama_index.vector_stores.lancedb import LanceDBVectorStore
 
-        vector_store = LanceDBVectorStore(uri="/tmp/lancedb")
+        vector_store = LanceDBVectorStore()  # native invocation
         ```
     """
 
     stores_text = True
     flat_metadata: bool = True
-    _connection: Any = PrivateAttr()
     uri: Optional[str]
-    table_name: Optional[str]
     vector_column_name: Optional[str]
     nprobes: Optional[int]
     refine_factor: Optional[int]
     text_key: Optional[str]
     doc_id_key: Optional[str]
+    api_key: Optional[str]
+    region: Optional[str]
+
+    _table_name: Optional[str] = PrivateAttr()
+    _connection: Any = PrivateAttr()
+    _table: Any = PrivateAttr()
+    _metadata_keys: Any = PrivateAttr()
 
     def __init__(
         self,
-        uri: Optional[str],
-        table_name: str = "vectors",
+        connection: Optional[Any] = None,
```

Comment:
will changing the order of args affect backwards compat? If so, let's try to add the new args towards the end?

Reply:
Yes, this was supposed to be a minor breaking change to synch with other integrations, thanks ill change it back.

Reply:
kk sounds good